### ISSUE TITLE
The documentation lacks of "getting started" part

### ISSUE THREAD

**Issue by loicknuchel** [mohiva/play-silhouette#385]
Hi,

I'm really new to silhouette and it seems a good auth lib but understanding how to setup all the thing the first time is quite difficult to me...
I've seen the issue #190 but as it's old, I create a new one.

What I need is a simple step-by-step guide on how to setup silhouette on my project for a simple use case (email auth + facebook/twitter).

Having sample applications is great but when you have an existing project it requires to make a diff and it's not easy to not miss any usefull file/line... Moreover, I'm using Play 2.3 and the sample application is only Play 2.4 & Silhouette 3...

So, I would be happy to contribute & write the getting started part of the doc but I need to understand how to do it first...

Any usefull link ?

**Comment by akkie** [comment]
Hi,

I know that the lack of a "Getting started guide" is really annoying. But I'm very busy with the development of Silhouette itself. I've noted before that this guide could be written by the community to support the project. So it would be really nice if you could take over this part. The seed template is tagged, so you can switch to version 2 of the template.

https://github.com/mohiva/play-silhouette-seed/tree/2.0

The basic steps are:
- Include the library
- Choose the appropriate authenticator
- Create your identity
- Create controllers based on the needed auth mechanism
- Write the services and models
- Wire dependencies

Maybe this tutorial can also help.
http://blog.knoldus.com/2015/05/11/play-framework-security-with-silhouette/

**Comment by loicknuchel** [comment]
Thanks for your quick answer.

I'll try to setup auth and if I feel confortable, i will write the getting started part.

Keep in touch

**Comment by ipsq** [comment]
I'd setup a getting started guide for you if I understood the actual process. At this point I'm still guessing what to implement and bind in my module. I can't get the basic credential provider working. Also, there are some things in the docs that don't work on my machine like the `#as` on the underlying config instance. Some basic insight on what I should implement would really contribute already. If I can get it working without a Database and just plain hardcoded users, I might be able to figure it out on my own.

**Comment by asazernik** [comment]
@akkie - if we do want to contribute to the docs, is the preferred method of doing so through the readme.io site where the docs are browsable? Or is there some separate github repository/directory where they're checked in?

**Comment by akkie** [comment]
@asazernik The getting started guide should be part of the documentation on readme.io. But I think it would make sense to create a Google docs document first so that it's easier to contribute for others too. What do you think?

**Comment by loicknuchel** [comment]
Yes, a shared google doc is a great solution to have the clearest getting started.

I have implemented auth using https://github.com/yzernik/silhouette-cake-seed as base exemple but it's with v1 :( I have to upgrade now (I don't use DI)

**Comment by asazernik** [comment]
Here's [a world-writable Google Doc](https://docs.google.com/document/d/1y5pLGFAxHmBhsWWgypXUam9nDU88BAXPTD8lNV8k4Cw/edit?usp=sharing). Right now all it has is my current notes from trying to piece together an implementation based on the docs and play-silhouette-seed; I'd very much appreciate input, especially that changes outline sections into readable prose.

**Comment by akkie** [comment]
@asazernik This is really great work! Now we should start to bring this notes in a readable format. @loicknuchel @ipsq Your help would be really appreciated!

I'll see that I can implement the default Guice module and the additional annotations.

**Comment by akkie** [comment]
@asazernik @loicknuchel @ipsq Any progress on this issue?

**Comment by asazernik** [comment]
Sorry, in kind of a crunch right now - hopefully I'll have spare time for this in a month or two, though I might sooner.

**Comment by MojoJojo** [comment]
A year later and I'm faced with the same issue around documentation. I would gladly take this job up. Can I perhaps submit a PR on the github wiki or a link to my own blog? Think it would be best to update the main documentation site though..

**Comment by akkie** [comment]
You could submit a chance request to the main documentation site. Maybe we could add a new section "Getting started" which can then be divided into different parts. It would be great if we do not have double content. So I think the preffered way would it to link to already existing documentation parts.

**Comment by Fruzenshtein** [comment]
What is the progress of current issue?

**Comment by loicknuchel** [comment]
Hi, I'm still having trouble figuring out how to use Silhouette with Play 2.5 and Compile time DI :(
I only want super simple setup using email & password.

Do somebody know a good resource to get started ? (when I figure out, I promise to do a blog article...)

**Comment by mig-foxbat** [comment]
+1. there is alot of room for improvements on documentation esp for those who are new to authenication. for eg, I had a question reg actions http://discourse.silhouette.rocks/t/request-types-in-silhouette-api/35
